js_modules/dagster-ui/packages/ui-core/src/workspace/WorkspaceContext/WorkspaceContext.tsx

```diff
@@ -198,14 +198,12 @@ export const WorkspaceProvider = ({children}: {children: React.ReactNode}) => {
     if (isRefetching) {
       return [];
     }
-    const toFetch = Object.values(locationStatuses).filter((loc) => {
-      const prev = prevLocationStatuses.current?.[loc.name];
-      const d = locationEntriesData[loc.name];
-      const entry = d?.__typename === 'WorkspaceLocationEntry' ? d : null;
+    const toFetch = Object.values(locationStatuses).filter((statusEntry) => {
+      const d = locationEntriesData[statusEntry.name];
+      const locationEntry = d?.__typename === 'WorkspaceLocationEntry' ? d : null;
       return (
-        prev?.updateTimestamp !== loc.updateTimestamp ||
-        prev?.loadStatus !== loc.loadStatus ||
-        entry?.loadStatus !== loc.loadStatus
+        locationEntry?.loadStatus !== statusEntry?.loadStatus ||
+        locationEntry?.updatedTimestamp !== statusEntry?.updateTimestamp
```

Comment:
oh interesting we were only comparing previous and current *status* timestamps before instead of comparing status timestamp against entry timestamp  - I could see how this would have caused things to get stuck

Im not even sure we need the load status check included here, the soon to be "version key" should be all thats required to know we should refetch but i dont think that can hurt

Reply:
yeah @alangenfeld that was my work around because the timestamps werent aligned before. But the indexeddb cache immediately saves the status query when in reality I would have had to save it after we fetched the updated location entries in order for this logic not to get stuck.